task/main.py

```diff
@@ -48,16 +48,26 @@ def discover(self):
         """
         Helper function to kick off the entire discover process
         """
-        granule_dict = self.discover_granules()
-        self.logger.info('##########')
-        self.logger.info(f'Discovered {len(granule_dict)} granules.')
-        self.check_granule_updates_db(granule_dict)
+        granules = self.collection.get('meta', {}).get('granules', None)
+        if granules:
+            self.logger.info(f'Received {len(granules)} to reingest.')
+            granule_dict = {}
+            for granule in granules:
+                self.populate_dict(granule_dict, key=granule, etag=None, last_mod=None)
```

Comment:
As an improvement: we don't pass a dictionary to a function (not something mutable) as a solution return that dictionary from  self.populate_dict to use it for self.cumulus_output_generator

Reply:
I'm not sure what is meant by this. Dictionaries are mutable so modifying them in this manner is a non-issue. What issue is there?